### PR TITLE
Added a versioning system for wisp.

### PR DESCRIPTION
**Pull request by VZout**
Added a versioning system that is fully constexpr and easy to increment.

### CODE REVIEW

File: src/version.hpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/*!
+A constexpr way to obtain program version information.
+Copyright © 2018, Viktor Zoutman
```
Comment (tntmeijs) [inline src/version.hpp]: ... and team? 😭

Comment (VZout) [reply src/version.hpp]: :| I wrote this code a years ago forgot to update the license.

File: src/version.hpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#include <cstdint>
+#include <cstdio>
+#include <stdexcept>
+
+namespace wr
+{
+
+	struct Version
+	{
+		std::uint32_t m_major = 0, m_minor = 0, m_patch = 0;
```
Comment (LaisoEmilio) [inline src/version.hpp]: I still don't like this, even if our code standard doesn't mention it :P your call though :)

File: src/version.hpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	};
+
+	namespace std_cexpr
+	{
+		constexpr size_t strlen(const char* str) {
```
Comment (LaisoEmilio) [inline src/version.hpp]: curly brackets in new line please :)

**Pull request merged**